packages/aws-cdk/test/integ/cli/cli.integtest.ts

```diff
@@ -518,6 +518,43 @@ integTest('cdk diff', withDefaultFixture(async (fixture) => {
     .rejects.toThrow('exited with error');
 }));
 
+integTest('enableDiffNoFail', withDefaultFixture(async (fixture) => {
```

Comment:
Kudos for the scenarios, by the time I was done reading them I got it, but it was a thoroughly confusing journey until then :-)

The option takes precedence in what the outcome is, unless it's not specified, and that is when the feature flag value matters.